torch/nn/functional.py

```diff
@@ -1153,8 +1153,36 @@ def batch_norm(input, running_mean, running_var, weight=None, bias=None,
     )
 
 
+def local_response_norm(input, size, alpha=1e-4, beta=0.75, k=1):
+    """Applies local response normalization over an input signal composed of
+    several input planes.
+
+    See :class:`~torch.nn.LocalResponseNorm` for details.
+    """
+    dim = input.dim()
+    if dim < 3 or dim > 5:
+        raise ValueError('Expected {3,4,5}D input (got {} dimensions)'
+                         .format(dim))
+    div = input.mul(input).unsqueeze(1)
+    if dim == 3:
+        div = pad(div, (0, 0, size // 2, (size - 1) // 2))
+        div = avg_pool2d(div, (size, 1), stride=1).squeeze(1)
+    elif dim == 4:
+        div = pad(div, (0, 0, 0, 0, size // 2, (size - 1) // 2))
+        div = avg_pool3d(div, (size, 1, 1), stride=1).squeeze(1)
+    elif dim == 5:
+        sizes = input.size()
+        div = div.view(sizes[0], 1, sizes[1], sizes[2], sizes[3] * sizes[4])
+        div = pad(div, (0, 0, 0, 0, size // 2, (size - 1) // 2))
+        div = avg_pool3d(div, (size, 1, 1), stride=1).squeeze(1)
+        div = div.view(sizes[0], sizes[1], sizes[2], sizes[3], sizes[4])
```

Comment:
hmmm the cases looks very similar and it should be quite easy to generalize this to support an arbitrary number of dimensions. Let me know if you want to do that (I think that's nicer), of if you'd rather get this in as is (I'm ok with merging the code as is)

Reply:
Yep done a generalised higher dim version. If someone can check the maths/implementation then this should be good to go.